app/forms/degree_form.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class DegreeForm
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+
+  FIELDS = %i[
+    slug
+    locale_code
+    uk_degree
+    non_uk_degree
+    subject
+    institution
+    graduation_year
+    grade
+    other_grade
+    country
+  ].freeze
+
+  attr_accessor(*FIELDS, :parent, :degree)
+
+  validate :validate_with_degree_model
+
+  delegate :uk?, :non_uk?, :non_uk_degree_non_naric?,
+           to: :degree
+
+  def initialize(parent:, degree:)
```

Comment:
Not sure about the keyword argument `parent`. `degrees_form`? At least you'll know what's actually coming in.